netbox/circuits/models/circuits.py

```diff
@@ -232,20 +236,22 @@ class CircuitTermination(
         choices=CircuitTerminationSideChoices,
         verbose_name=_('termination')
     )
-    site = models.ForeignKey(
-        to='dcim.Site',
+    scope_type = models.ForeignKey(
+        to='contenttypes.ContentType',
         on_delete=models.PROTECT,
-        related_name='circuit_terminations',
+        limit_choices_to=Q(model__in=CIRCUIT_TERMINATION_SCOPE_TYPES),
+        related_name='+',
         blank=True,
         null=True
     )
-    provider_network = models.ForeignKey(
-        to='circuits.ProviderNetwork',
-        on_delete=models.PROTECT,
-        related_name='circuit_terminations',
+    scope_id = models.PositiveBigIntegerField(
         blank=True,
         null=True
     )
+    scope = GenericForeignKey(
```

Comment:
I'm not sure it makes sense to call this "scope;" it's more of a termination point. `termination` might be more apt, and it would be consistent with the CableTermination model.